projects/js-packages/connection/state/resolvers.jsx

```diff
@@ -24,6 +24,10 @@ const connectionResolvers = {
 				args
 			);
 
+			if ( ! hasAuthorization ) {
```

Comment:
This way `invalidateResolution` will be called in moments that we don't have the authorization and didn't finish the resolution yet.

```suggestion
                        if ( ! hasAuthorization && hasFinishedResolution ) {
```